presto-main-base/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -3022,6 +3032,20 @@ public static List<PlanCanonicalizationStrategy> getHistoryOptimizationPlanCanon
         return strategyList;
     }
 
+    public static List<QueryType> getQueryTypesEnabledForHBO(Session session)
+    {
+        List<QueryType> queryTypes;
+        try {
+            queryTypes = Splitter.on(",").trimResults().splitToList(session.getSystemProperty(QUERY_TYPES_ENABLED_FOR_HISTORY_BASED_OPTIMIZATION, String.class)).stream()
+                    .map(QueryType::valueOf).collect(toImmutableList());
+        }
+        catch (Exception e) {
+            queryTypes = ImmutableList.of();
```

Comment:
we should throw an exception instead for invalid query types like we do for other invalid configs.